app/user/rest.py

```diff
@@ -178,6 +178,17 @@ def activate_user(user_id):
     return jsonify(data=user.serialize()), 200
 
 
+@user_blueprint.route("/<uuid:user_id>/deactivate", methods=["POST"])
+def deactivate_user(user_id):
+    user = get_user_by_id(user_id=user_id)
+    if user.state == "pending":
```

Comment:
We don't have an `enum` to reference here for this, right?  I only see an `enum` for [`InvitedUserStatus`](https://github.com/GSA/notifications-api/blob/main/app/enums.py#L182) and this isn't referring to an `InvitedUser`, right?

It caught my eye though because it seems a bit odd that the status would be `pending` unless that is what we're referring to.  If that's the case, we may be leveraging the `InvitedUser` plumbing here when instead we just want to swap an actual user from being actively signed in vs. not, and I worry that this deactivation will actually tap into user invite mechanism and could then put their account in a weird state.

I might be overthinking it, but at the same time if this is the case we'll need to test to make sure.  Ideally there ought to be another way where we are only activating/deactivating existing full users and not triggering a state change that could also affect how the app manages the account and possibly flips them back to being an `InvitedUser` (especially if it does so just partially).

In a broader sense, whether or a user account is `active` or `inactive` _should_ be entirely separate from whether or not they're logged in or not...  most other apps track the two states entirely separately, and flip users who haven't logged in for a period of time (e.g., 90 days) to being inactive automatically and the next time a user signs in, it either walks them through reactivating their account or let's them know they need to reach out for app support to get their account activated/enabled again.  I realize this app functions a bit differently around this notion, but I wanted to call out the fact that we're operating under a slightly different set of assumptions and criteria when it comes to this.

Reply:
activate() is called every single time the user logs in, and that is pre-existing code from notify UK.  So, for this issue, I may be overloading the invite active/pending state as you suggest, but functionally it's immaterial because you can log in out and out a dozen times and it just toggles the state without firing off emails or making you ineligible.

So I'm happy to create new states if you like.  They don't exist currently and there is no way to track if a user is logged in or not (currently).

Reply:
Okay, this makes sense to me and I was definitely overthinking given the current state of things.  Thanks for explaining and clarifying the current login/logout state management!

If all this is doing is simply toggling the state that gets reset every time you log back in anyway, then this is fine for our needs now.